common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java

```diff
@@ -46,6 +46,43 @@
 
   protected volatile TransportClientFactory clientFactory;
   protected String appId;
+  protected TransportConf transportConf;
```

Comment:
Thoughts on making this `final` and initialize it in constructor ?

Reply:
I tried to make all these three fields initialize in a constructor (it seems a bit werid to me if we only try on `transportConf`) but failed due to some dependency issue (because `clientFactory` can only be initialized after `init()`). And there's not a few changes if we want to initialize `transportConf ` in a constructor. So I finally decide to leave it as is.

Reply:
Yeah, I was not sure if this would be easy to do ... wanted to surface it anyway. Thanks for double checking @Ngone51 !